skyportal/handlers/api/invitations.py

```diff
@@ -102,114 +102,122 @@ def post(self):
+            role_id = data.get("role", "Full user")
+            if role_id not in ["Full user", "View only"]:
+                return self.error(
+                    f"Unsupported value provided for parameter `role`: {role_id}. "
+                    "Must be one of either 'Full user' or 'View only'."
+                )
+            role = session.scalars(
+                Role.select(self.current_user).where(Role.id == role_id)
+            ).first()
 
-        if data.get("groupIDs") is None:
-            return self.error("Missing required parameter `groupIDs`")
-        try:
-            group_ids = [int(gid) for gid in data["groupIDs"]]
-        except ValueError:
-            return self.error(
-                "Invalid value provided for `groupIDs`; unable to parse "
-                "all list items to integers."
-            )
-        groups = Group.get_if_accessible_by(group_ids, self.current_user)
-        if set(group_ids).difference({g.id for g in groups}):
-            return self.error(
-                "The following groupIDs elements are invalid: "
-                f"{set(group_ids).difference({g.id for g in groups})}"
-            )
+            if data.get("userEmail") in [None, "", "None", "null"]:
+                return self.error("Missing required parameter `userEmail`")
+            user_email = data["userEmail"].strip()
 
-        if data.get("streamIDs") not in [None, "", "null", "None"]:
+            if data.get("groupIDs") is None:
+                return self.error("Missing required parameter `groupIDs`")
             try:
-                stream_ids = [int(sid) for sid in data["streamIDs"]]
+                group_ids = [int(gid) for gid in data["groupIDs"]]
             except ValueError:
                 return self.error(
-                    "Invalid value provided for `streamIDs`; unable to parse "
+                    "Invalid value provided for `groupIDs`; unable to parse "
                     "all list items to integers."
                 )
-
-            streams = Stream.get_if_accessible_by(stream_ids, self.current_user)
-            if set(stream_ids).difference({s.id for s in streams}):
+            groups = session.scalars(
+                Group.select(self.current_user).where(Group.id.in_(group_ids))
+            ).all()
+            if set(group_ids).difference({g.id for g in groups}):
                 return self.error(
-                    "The following streamIDs elements are invalid: "
-                    f"{set(stream_ids).difference({s.id for s in streams})}"
+                    "The following groupIDs elements are invalid: "
+                    f"{set(group_ids).difference({g.id for g in groups})}"
                 )
 
-            # Ensure specified groups are covered by specified streams
-            if not all(
-                [stream in streams for group in groups for stream in group.streams]
-            ):
+            if data.get("streamIDs") not in [None, "", "null", "None"]:
```

Comment:
We really need a utility function for this check, since it appears so often; not necessary for this PR.